File: bench/py/35_map_filter.py
# map() then filter() with lambdas (per-element function calls, like MyLang)
import sys
scale = int(sys.argv[1]) if len(sys.argv) > 1 else 1

SZ = 500000 * scale
a = list(range(SZ))

b = list(map(lambda x: x * 2, a))
c = list(filter(lambda x: x % 3 == 0, b))

print("result:", len(c))

File: bench/ml/35_map_filter.ml
# map() then filter() over a large array, each calling a lambda per element
var scale = 1;
if (len(argv) > 0)
    scale = int(argv[0]);

var SZ = 500000 * scale;
var a = array(SZ);
for (var i = 0; i < SZ; i += 1)
    a[i] = i;

var b = map(func(x) => x * 2, a);
var c = filter(func(x) => x % 3 == 0, b);

print("result:", len(c));
